### ISSUE TITLE
Snippet state is not updated when built-in completion popup menu is visible

### ISSUE THREAD

**Issue by echasnovski** [neovim/neovim#25895]
### Problem

When completion popup menu is visible during entering tabstop text but no its entry is selected, snippet state seems to be not updated properly. This situation is common with autocompletion which uses built-in completion or if indeed user wants to use on demand built-in completion.

### Steps to reproduce

1. Create the following config file:
```lua
vim.keymap.set('i', '<C-l>', function() vim.snippet.jump(1) end)
vim.keymap.set('i', '<C-e>', function() vim.snippet.expand('$1 = function($2)\n$3\nend') end)
```
2. Start Neovim with `--clean` and this config file.
3. Start Insert mode and press `<C-e>` to expand the snippet.
4. Press `<C-n>` to show completion popup and `<C-p>` to make no item selected.
5. Press `<C-l>` to jump forward.
6. Type any character. Whole line up until cursor is highlighted as if entered text in tabstop. Pressing `<C-l>` again moves cursor to the start of the snippet.

### Expected behavior

Snippet navigation does not depend on whether popup is visible or not.

------

What I assume is that some state tracking is not triggered when popup menu is visible. There are several places which met the eye on the first sight:
- [These autocommands](https://github.com/neovim/neovim/blob/5cefec7349610853910c21a0215f85a4d47132d1/runtime/lua/vim/snippet.lua#L322) don't include `TextChangedP` (as `TextChangedI` is "Not triggered when the popup menu is visible."). Although, I've added it and didn't see positive changes, but maybe I am missing something.
- [These autocommands](https://github.com/neovim/neovim/blob/5cefec7349610853910c21a0215f85a4d47132d1/runtime/lua/vim/snippet.lua#L276) are not triggered when popup menu is visible (as per `:h CursorMovedI`). I would assume those should not be crucial here because cursor move is done with `vim.snippet.jump()`, but I might be missing something.

### Neovim version (nvim -v)

NVIM v0.10.0-dev-1466+gec66a95fb-dirty

### Vim (not Nvim) behaves the same?

No

### Operating system/version

EndeavourOS Linux x86_64 (6.5.9-arch2-1)

### Terminal name/version

st-0.9

### $TERM environment variable

st-256color

### Installation

From source

**Comment by MariaSolOs** [comment]
@echasnovski I can no longer reproduce this (might have been fixed with #26879). Could you please try it out and lmk?

**Comment by echasnovski** [comment]
Yes, I can no longer reproduce this also.

Because #26879 does not look like fixing this, I got curious and bisected. This was fixed in d6f429520bce3c9f5d97f920e847fe6f668fac79 (which is part of #25795). And the real culprit fixing this looks like [this `if`](https://github.com/neovim/neovim/blob/eb1e8c12e2c5ad172f2fe54cb96bf31602b16c3d/runtime/lua/vim/snippet.lua#L292-L295). Not sure if `vim.fn.complete(vim.fn.col('.'), {})` is a good way to "close completion menu" (I'd prefer something like `vim.fn.feedkeys('\25', 'n')`, which is for "stop completion and accept the currently selected entry"), but I don't have any objective reason to back this up.

Having some kind of test against repeating this would be a good idea, though. This issue originally was quite intrusive as it meant that user could not expand snippet when completion menu is visible.
